### ISSUE TITLE
Support "none" rule

### ISSUE THREAD

**Issue by ccoVeille** [loeffel-io/ls-lint#234]
I woulike to have a way to use this without having to play with multiple rules or using an hack

Here is an example to consider:

- enforce all directories and subdirectories to be lowercase, except one
- but one folder can contain any case/format, but not their sub-directories

Here is a tree

```
styles/
             fooBar/
                           testsa
             Foo_bar/
                           testsb
             foo-bar
                            testsc
testsd
```

I cannot use ignore, because I want the YAML files in each folder to be linted.

I would like to be able to say

```yaml
ls:
    .dir: lowercase
    .yaml: PascalCase
   
    styles/*:
       .dir: none
```

Right now, I think I can use `regexp: .` but it sounds like an hack.

It's pure pseudo-code. I faced this issues by working on

https://github.com/jargonLint/jargonLint/tree/main

I'm reporting  this issue from my phone while I temporarily have no computer with me

**Comment by loeffel-io** [comment]
Duplicate of exists which is already wip? 

https://github.com/loeffel-io/ls-lint/pull/223

**Comment by ccoVeille** [comment]
I'm unsure it would provide the thing I described, but maybe. The PR is huge, I don't think "exists" will solve my issue, but maybe you are referring to another feature you are adding via this PR

I will wait for the changes to be merged.

There is no rush for me.

**Comment by loeffel-io** [comment]
> I'm unsure it would provide the thing I described, but maybe. The PR is huge, I don't think "exists" will solve my issue, but maybe you are referring to another feature you are adding via this PR
> 
> I will wait for the changes to be merged.
> 
> There is no rush for me.

this should be your use case? 

![Bildschirmfoto 2024-07-11 um 09 55 13](https://github.com/loeffel-io/ls-lint/assets/12670730/455864ae-32cf-495c-b2c9-d5bbf5bc6f7c)

**Comment by loeffel-io** [comment]
got it - this is not supported by design. it should be instantly clear which rules are applied to which extension/directory

**Comment by ccoVeille** [comment]
> got it - this is not supported by design. it should be instantly clear which rules are applied to which extension/directory 

OK thanks

I will keep using a dirty hack like using regexp . for yaml file is that folder

**Comment by ccoVeille** [comment]
Please switch it back to not planned

**Comment by loeffel-io** [comment]
This feature is now available in the `v2.3.0-beta.1` release. 
Please test it and let me know if there are any issues ❤️ 

The upcoming v2.3.0 blog post: https://ls-lint.org/blog/announcements/v2.3.0.html
The docs: https://ls-lint.org/2.3/getting-started/introduction.html
The release notes: https://github.com/loeffel-io/ls-lint/releases/tag/v2.3.0-beta.1
